app/src/main/java/io/apicurio/registry/services/auth/BearerTokenExtractor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.services.auth;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.keycloak.authorization.client.AuthzClient;
+import org.keycloak.authorization.client.Configuration;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+//Extracts username, password pair from header if exists and request an access token to keycloak using those credentials
+//Once Quarkus support both basic and bearer working at the same time against Keycloak, we can remove this
+public class BearerTokenExtractor {
+
+    private static final String BASIC = "basic";
+    private static final String BASIC_PREFIX = BASIC + " ";
+    private static final String LOWERCASE_BASIC_PREFIX = BASIC_PREFIX.toLowerCase(Locale.ENGLISH);
+    private static final int PREFIX_LENGTH = BASIC_PREFIX.length();
+    private static final Charset charset = StandardCharsets.UTF_8;
+    private final AuthzClient keycloakClient;
+    private final RoutingContext context;
+
+    public BearerTokenExtractor(RoutingContext context, final String authServerUrl, String realmName, String clientId, String clientSecret) {
+        this.context = context;
+        final HashMap<String, Object> credentials = new HashMap<>();
+        credentials.put("secret", clientSecret);
+        final Configuration keycloakConfiguration = new Configuration(authServerUrl, realmName, clientId, credentials, null);
+        this.keycloakClient = AuthzClient.create(keycloakConfiguration);
```

Comment:
I think this will cause issues with native images, maybe we should use here the new client you are building

Reply:
Yes, this will be replaced by the new code present in the client-common repo. This is a refactor from already existing code to unblock things quickly and, afaik, we're not using native images in mas, right? This will be fixed in a short timeframe.